### PR TITLE
Update operator-sdk to 1.9.0

### PR DESCRIPTION
**Pull request by oribon**

### CODE REVIEW

**Review by SchSeba — COMMENTED** [review]
/ok-to-test

File: Makefile [hunk 1]
```diff
@@ -28,12 +226,9 @@ functests:
 generate-client:
 	bash vendor/k8s.io/code-generator/generate-groups.sh client \
```
Comment (SchSeba) [inline Makefile:227]: This file doesn't exist in the vendor folder any more

File: Makefile [hunk 2]
```diff
@@ -1,24 +1,222 @@
...
 	hack/build.sh
+
 image:
 	docker build -t openshift.io/ptp-operator -f Dockerfile.rhel7 .
+
 clean:
 	rm -rf build/_output/bin/ptp-operator
+
 deploy-setup:
 	hack/deploy-setup.sh
```
Comment (SchSeba) [inline Makefile:218]: you will need to update this script I think with the new config

File: Makefile [hunk 3]
```diff
@@ -1,24 +1,222 @@
...
+	
 undeploy:
 	hack/undeploy.sh
```
Comment (SchSeba) [inline Makefile:221]: you will need to update this script I think with the new config

File: Makefile [hunk 4]
```diff
@@ -1,24 +1,222 @@
...
+
+##@ Deployment
+
+install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build config/crd | kubectl apply -f -
+
+uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build config/crd | kubectl delete -f -
+
+deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
```
Comment (SchSeba) [inline Makefile:116]: please check this works to deploy

File: Makefile [hunk 5]
```diff
@@ -1,24 +1,222 @@
...
+
+##@ Build
+
+build: generate fmt vet ## Build manager binary.
+	go build -o bin/manager main.go
+
+run: manifests generate fmt vet ## Run a controller from your host.
+	go run ./main.go
+
+docker-build: test ## Build docker image with the manager.
```
Comment (SchSeba) [inline Makefile:102]: I don't think this will work as I don't see a dockerfile

File: Makefile [hunk 6]
```diff
@@ -1,24 +1,222 @@
...
+	docker build -t ${IMG} .
+
+docker-push: ## Push docker image with the manager.
```
Comment (SchSeba) [inline Makefile:105]: I don't think this will work as I don't see a dockerfile

File: Makefile [hunk 7]
```diff
@@ -1,24 +1,222 @@
...
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+fmt: ## Run go fmt against code.
+	go fmt ./...
+
+vet: ## Run go vet against code.
+	go vet ./...
+
+ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
+test: manifests generate fmt vet ## Run tests.
```
Comment (SchSeba) [inline Makefile:89]: this work?

File: build/Dockerfile [hunk 8]
```diff
@@ -1,16 +0,0 @@
-FROM registry.access.redhat.com/ubi7/ubi-minimal:latest
```
Comment (SchSeba) [inline build/Dockerfile:1]: I think we should allow the user to deploy the operator as the dockerfile.rhel7 is for internal build

File: config/rbac/kustomization.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+# subjects if changing service account names.
+- service_account.yaml
+- role.yaml
+- role_binding.yaml
+- leader_election_role.yaml
+- leader_election_role_binding.yaml
+# Comment the following 4 lines if you want to disable
+# the auth proxy (https://github.com/brancz/kube-rbac-proxy)
+# which protects your /metrics endpoint.
+- auth_proxy_service.yaml
```
Comment (SchSeba) [inline config/rbac/kustomization.yaml:15]: I don't think we need this

File: controllers/suite_test.go [hunk 10]
```diff
@@ -0,0 +1,83 @@
+/*
```
Comment (SchSeba) [inline controllers/suite_test.go:1]: this file was not here before I think we can remove this for now and work on unit test later

File: Makefile [hunk 11]
```diff
@@ -1,24 +1,222 @@
...
 bin:
 	hack/build.sh
+
 image:
 	docker build -t openshift.io/ptp-operator -f Dockerfile.rhel7 .
+
 clean:
 	rm -rf build/_output/bin/ptp-operator
+
 deploy-setup:
```
Comment (SchSeba) [inline Makefile:217]: I think we need to remove the deploy folder and use the config one

File: controllers/ptpconfig_controller.go [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=ptp.openshift.io,resources=ptpconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ptp.openshift.io,resources=ptpconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=ptp.openshift.io,resources=ptpconfigs/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
```
Comment (SchSeba) [inline controllers/ptpconfig_controller.go:50]: please update the comment

File: controllers/ptpoperatorconfig_controller.go [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+	ResyncPeriod = 2 * time.Minute
+)
+
+//+kubebuilder:rbac:groups=ptp.openshift.io,resources=ptpoperatorconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ptp.openshift.io,resources=ptpoperatorconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=ptp.openshift.io,resources=ptpoperatorconfigs/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
```
Comment (SchSeba) [inline controllers/ptpoperatorconfig_controller.go:62]: please update the comment

**Review by SchSeba — COMMENTED** [review]
looks good just some comments

File: main.go [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	namespace := os.Getenv("WATCH_NAMESPACE")
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
+		Port:               9443,
+		LeaderElection:     enableLeaderElection,
```
Comment (SchSeba) [inline main.go:71]: we need to enable leader election

File: main.go [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+		LeaderElectionID:   "c5793dcc.openshift.io",
```
Comment (SchSeba) [inline main.go:72]: change this to ptp.openshift.io

File: bundle/metadata/annotations.yaml [hunk 16]
```diff
@@ -0,0 +1,14 @@
+annotations:
+  # Core bundle annotations.
+  operators.operatorframework.io.bundle.mediatype.v1: registry+v1
+  operators.operatorframework.io.bundle.manifests.v1: manifests/
+  operators.operatorframework.io.bundle.metadata.v1: metadata/
+  operators.operatorframework.io.bundle.package.v1: ptp-operator
+  operators.operatorframework.io.bundle.channels.v1: alpha
```
[the remark this answers is not in the document]
Comment (oribon) [reply bundle/metadata/annotations.yaml:7]: It is auto-generated by make bundle. If I change it gets overridden to alpha.

File: bundle/manifests/ptp-operator.clusterserviceversion.yaml [hunk 17]
```diff
@@ -0,0 +1,344 @@
...
+              name: ptp-operator
+          strategy: {}
+          template:
+            metadata:
+              labels:
+                name: ptp-operator
+            spec:
+              containers:
+              - command:
+                - ptp-operator
```
[the remark this answers is not in the document]
Comment (oribon) [reply bundle/manifests/ptp-operator.clusterserviceversion.yaml:249]: It is the same as the csv in art manifests/ , I'll verify it works and update

File: config/custom/kustomization.yaml [hunk 18]
```diff
@@ -0,0 +1,2 @@
+resources:
```
[the remark this answers is not in the document]
Comment (oribon) [reply config/custom/kustomization.yaml:1]: That way kustomize knows about the daemon needed rbacs and deploys them (I also changed stuff in the Makefile deploy/bundle)

File: config/manager/manager.yaml [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+        operator: Exists
+      serviceAccountName: ptp-operator
+      containers:
+        - name: ptp-operator
+          # Replaces this with the image name
+          image: controller
```
[the remark this answers is not in the document]
Comment (oribon) [reply config/manager/manager.yaml:34]: it is replaced by kustomize, we run $(KUSTOMIZE) edit set image controller=${IMG}

File: main.go (hunk 14 above)
Comment (oribon) [reply main.go:71]: this should always be true?

File: hack/deploy-setup.sh [hunk 20]
```diff
@@ -5,9 +5,9 @@ pushd ${REPO_DIR}/deploy
 		envsubst< namespace.yaml | ${OPERATOR_EXEC} apply -f -
 	fi
 
-	FILES="crds/ptp.openshift.io_nodeptpdevices_crd.yaml
-		crds/ptp.openshift.io_ptpoperatorconfigs_crd.yaml
-		crds/ptp.openshift.io_ptpconfigs_crd.yaml
+	FILES="../config/crd/bases/ptp.openshift.io_nodeptpdevices.yaml
```
[the remark this answers is not in the document]
Comment (oribon) [reply hack/deploy-setup.sh:8]: I think we can ditch this script at this point, make deploy/undeploy looks good

File: hack/undeploy.sh [hunk 21]
```diff
@@ -8,9 +8,9 @@ pushd ${REPO_DIR}/deploy
 		clusterrole.yaml
 		role.yaml
 		rolebinding.yaml
-		crds/ptp.openshift.io_nodeptpdevices_crd.yaml
-		crds/ptp.openshift.io_ptpoperatorconfigs_crd.yaml
-		crds/ptp.openshift.io_ptpconfigs_crd.yaml"
+		../config/crd/bases/ptp.openshift.io_nodeptpdevices.yaml
```
[the remark this answers is not in the document]
Comment (oribon) [reply hack/undeploy.sh:11]: ^^

File: bundle/manifests/ptp-operator.clusterserviceversion.yaml (hunk 17 above)
[the remark this answers is not in the document]
Comment (oribon) [reply bundle/manifests/ptp-operator.clusterserviceversion.yaml:249]: it works like this

File: Dockerfile [hunk 22]
```diff
@@ -0,0 +1,18 @@
+FROM golang:1.15 AS builder
+WORKDIR /go/src/github.com/openshift/ptp-operator
+COPY . .
+ENV GO111MODULE=off
+RUN make
+
+FROM quay.io/openshift/origin-base:4.7
+COPY --from=builder /go/src/github.com/openshift/ptp-operator/build/_output/bin/ptp-operator /usr/local/bin/
```
[the remark this answers is not in the document]
Comment (oribon) [reply Dockerfile:8]: this way it is aligned with the manifests/ and it seems to work

File: main.go (hunk 14 above)
Comment (aneeshkp) [reply main.go:71]: getting his error when I deploy 
```
unknown flag: --enable-leader-election

```

Comment (aneeshkp) [reply main.go:71]: ignore my comment. I was using an old binary.

**Pull request merged**